sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -560,13 +567,24 @@ class StreamExecution(
     }
     if (hasNewData) {
       var batchWatermarkMs = offsetSeqMetadata.batchWatermarkMs
-      // Update the eventTime watermark if we find one in the plan.
+      // Update the eventTime watermarks if we find any in the plan.
```

Comment:
its still a single watermark that is being updated. it just happens to be updated using multiple watermarks.

Reply:
Well, we're updating multiple watermarks in the map. We later update `offsetSeqMetadata` with the new minimum one, but that's not in this block.

Reply:
Yeah. I had made this comment when i was thinking that we dont need the mutable map. Ignore this.